### PR TITLE
Split MIRK methods into subpackage

### PR DESCRIPTION
**Pull request by ErikQQY**
Superseded #212, let's split one by one

### CODE REVIEW

**Comment by ErikQQY** [comment]
The benchmarking CI failure is expected since it's comparing the master branch and this branch which contains the subpackage that doesn't register yet. Once the subpackage is registered, the benchmarking CI will be fine.

**Comment by ErikQQY** [comment]
Bump, this one is ready to go

File: Project.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ LinearSolve = "7ed4a6bd-45f5-4d41-b270-4a48e9bafcae"
 Logging = "56ddb016-857b-54e1-b83d-db4d58db5568"
 NonlinearSolve = "8913a72c-1f9b-4ce2-8d82-65094dcecaec"
 OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
+Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
```
Comment (avik-pal) [inline Project.toml:20]: do you need it in main deps?

**Review by ChrisRackauckas — APPROVED** [review]

File: Project.toml (hunk 1 above)
Comment (ErikQQY) [reply Project.toml]: Just need Pkg when running the tests, just fixed

**Pull request merged**